packages/ui/src/components/DatePicker/SDatePicker.vue

```diff
@@ -0,0 +1,734 @@
+              class="options-panel sora-tpg-p3"
+            >
+              <OptionsWrapper v-slot="{ options }">
+                <p
+                  v-for="(item, idx) in options[props.type]"
+                  :key="idx"
+                  class="options-panel__item"
+                  :class="menuState === item.label ? 'active' : ''"
+                  @click="onMenuClick(item.value, item.label)"
+                  @keydown="onMenuClick(item.value, item.label)"
+                >
+                  {{ item.label }}
+                  <span
+                    v-show="menuState === item.label"
+                    class="options-panel__checkmark"
+                  ><IconBasicCheckMark24 /></span>
+                </p>
+              </OptionsWrapper>
+            </div>
+            <div class="calendars-panel">
+              <div>
+                <DatePanel
+                  v-show="currentView === 'dates'"
+                  :show-state="showState"
+                  :value="firstCalendarModelValue"
+                  :time="props.time"
+                  :selection-mode="type"
+                  :state-store="stateStore"
+                  @change-view="changeView"
+                  @update-showed-state="updateShowedState"
+                  @pick="onDatePick"
+                />
+              </div>
+              <div
+                v-if="isRange"
+                class="ml-4"
+              >
+                <DatePanel
+                  v-show="currentView === 'dates'"
+                  :show-state="nextMonthShowState"
+                  :value="calendarToModelValue"
+                  :time="props.time"
+                  :selection-mode="type"
+                  :state-store="stateStore"
+                  :hide-arrows="true"
+                  @change-view="changeView"
+                  @update-showed-state="updateShowedState"
+                  @pick="onDatePick"
+                />
+              </div>
+              <div
+                v-if="currentView === 'months'"
+                class="h-full w-full flex items-start justify-center"
+              >
+                <MonthTable
+                  :show-state="showState"
+                  @pick="updateShowedMonth"
+                  @update-showed-year="updateShowedYear"
+                />
+              </div>
+              <div v-if="currentView === 'years'">
+                <YearTable
+                  :value="showState.year"
+                  @pick="updateShowedYear"
+                />
+              </div>
+            </div>
+            <div
+              v-if="time && !showStateView"
+              class="time-panel"
+            >
+              <TimePanel
+                :value="currentValueTime"
+                @update-time="updateTime"
+              />
+            </div>
+            <div
+              v-if="menuState === 'Custom' && !showStateView"
+              class="custom-panel"
+            >
```

Comment:
Same as for `option-panel` - extract it please into something like `SDatePickerCustomPanel.vue`.

By the way, if there will be `...CustomPanel` and `OptionPanel` components, it's better to place `Panel` word earlier. Maybe `PanelCustom` and `PanelBase`?